src/NDItk/MultigroupTable.hpp

```diff
@@ -0,0 +1,88 @@
+#ifndef NJOY_NDITK_MULTIGROUP
+#define NJOY_NDITK_MULTIGROUP
+
+// system includes
+
+// other includes
+#include "tools/Log.hpp"
+#include "NDItk/multigroup/Metadata.hpp"
+#include "NDItk/multigroup/Structure.hpp"
+#include "NDItk/multigroup/FluxWeights.hpp"
+#include "NDItk/multigroup/ReactionCrossSections.hpp"
+
+namespace njoy {
+namespace NDItk {
+
+/**
+ *  @brief A multigroup NDI table
+ */
+class MultigroupTable {
+
+  /* fields */
+
+  multigroup::Metadata metadata_;
+  multigroup::Structure structure_;
+  multigroup::FluxWeights weights_;
+  multigroup::ReactionCrossSections xs_;
+
+  /* auxiliary functions */
+
+  #include "NDItk/MultigroupTable/src/readRecord.hpp"
+  #include "NDItk/MultigroupTable/src/verify.hpp"
+
+public:
+
+  /* constructor */
+
+  #include "NDItk/MultigroupTable/src/ctor.hpp"
+
+  /* methods */
+
+  /**
+   *  @brief Return the metadata of the table
+   */
+  const multigroup::Metadata& metadata() const { return this->metadata_; }
+
+  /**
+   *  @brief Return the primary group structure record
+   */
+  const multigroup::Structure& structure() const { return this->structure_; }
```

Comment:
C++ still has structs ;-)

Consider it changed. This is a major part of what reviews are for: I need someone to look at the API and be like "this could be better" :-)

Reply:
As a related question: should we change Structure to GroupStructure too (similar to FluxWeights)?

Reply:
Yes.

Reply:
Work style question: are we marking comments as resolve prior to them being implemented? Should we not push an update to the branch with the PR prior to marking this a resolved?

Reply:
Depends, if it is the only outstanding PR then we just make the modifications as additional commits to the PR. In this case, this is already 3 PRs behind (we have had even worse). For cases like these, we often use a separate update branch or refer to other PRs where things were changed (or where the issue became obsolete).